app/assets/src/components/Samples.jsx

```diff
@@ -938,53 +923,31 @@ class Samples extends React.Component {
     return !this.state.columnsShown.includes(column);
   }
 
-  initializeSelectAll() {
-    // select all checkboxes
-    var that = this;
-    $(".checkAll").click(function(e) {
-      var checked = e.currentTarget.checked;
-      $(".checkbox:enabled").prop("checked", checked);
-      that.setState({
-        allChecked: checked
-      });
-      that.fetchAllSelectedIds(checked);
-    });
-  }
-
-  checkTheRightBoxes() {
-    var that = this;
-    $(".checkbox:enabled").each((id, element) => {
-      let sample_id = element.getAttribute("data-sample-id");
-      const sampleList = that.state.selectedSampleIds;
-      if (!sample_id) {
-        return;
+  fetchAllSelectedIds(e) {
+    let sampleList = this.state.selectedSampleIds;
+    const checked = e.target.checked;
+    const allSamples = this.state.allSamples;
+    for (let idx in allSamples) {
+      let sample = allSamples[idx];
+      if (sample.run_info.report_ready != 1) {
+        continue;
       }
-      let i = parseInt(sample_id);
-      if (sampleList.indexOf(i) >= 0) {
-        element.checked = true;
-      } else {
-        element.checked = false;
-      }
-    });
-  }
-
-  fetchAllSelectedIds(checked) {
-    var that = this;
-    let sampleList = that.state.selectedSampleIds;
-    $(".checkbox:enabled").each((id, element) => {
-      let sample_id = parseInt(element.getAttribute("data-sample-id"));
-      let indx = sampleList.indexOf(sample_id);
+      let sample_id = sample.db_sample.id;
       if (checked) {
-        if (indx === -1) {
-          sampleList.push(+sample_id);
+        if (sampleList.indexOf(sample_id) === -1) {
```

Comment:
You might want to consider making sampleList a Set object instead so you can use functions like has and delete instead of indexOf != -1 and indexOf/splice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Babel provides a polyfill to provide full compatibility with older browsers.

Reply:
Thanks for the tip.